### ISSUE TITLE
Build a proof of concept of the idea locality-based routing using Envoy

### ISSUE THREAD

**Issue by allenlsy** [openservicemesh/osm#3557]
**Please describe the Improvement and/or Feature Request**

This is a subtask of #3043 .  Refer to #3043 for the details of the POC.

**Scope (please mark with X where applicable)**

| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | ✅ |
| Ingress                    | [ ] |
| Egress                     | ✅ |
| Networking                 | ✅ |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

**Possible use cases**

**Comment by allenlsy** [comment]
Here is a simple POC demo for this issue:

https://github.com/allenlsy/envoyproxy-locality-weighted-lb-demo

TLDR is, Envoy locality weight load balancing works as described in its documentation. By default when the overprovision rate is lower than 1.4, which means less than 71% of local workloads are healthy, Envoy starts to split traffic to other localities, based on weight and priority.

**Comment by allenlsy** [comment]
cc @snehachhabria @delqn @shashankram
